mlrun/utils/helpers.py

```diff
@@ -667,6 +674,29 @@ def enrich_image_url(image_url: str, client_version: str = None) -> str:
     return image_url
 
 
+def resolve_image_tag_suffix(
+    mlrun_version: str = None, python_version: str = None
+) -> str:
+    """
+    resolves what suffix should be appended to the image tag
+    :param mlrun_version: the mlrun version
+    :param python_version: the requested python version
+    :return: the suffix to append to the image tag
+    """
+    if not python_version or not mlrun_version:
+        return ""
+    # For mlrun 1.3.0, we decided to support mlrun runtimes images with both python 3.7 and 3.9 images.
+    # While the python 3.9 images will continue to have no suffix, the python 3.7 images will have a '-py37' suffix.
+    # Python 3.8 images will not be supported for mlrun 1.3.0, meaning that if the user has client with python 3.8
+    # and mlrun 1.3.x then the image will be pulled without a suffix (which is the python 3.9 image).
+    # using semver (x.y.z-X) to include rc versions as well
+    if semver.VersionInfo.parse("1.3.0-X") <= semver.VersionInfo.parse(
+        mlrun_version
+    ) < semver.VersionInfo.parse("1.4.0-X") and python_version.startswith("3.7"):
```

Comment:
Same as other comments - limiting to 1.4.0 seems premature, we might still support py37 there as well.